packages/l/libvorbis/xmake.lua

```diff
@@ -10,13 +10,71 @@ package("libvorbis")
 
     add_versions("1.3.7", "0e982409a9c3fc82ee06e08205b1355e5c6aa4c36bca58146ef399621b0ce5ab")
 
+    add_configs("vorbisenc",  {description = "Includes vorbisenc", default = true, type = "boolean"})
+    add_configs("vorbisfile", {description = "Includes vorbisfile", default = true, type = "boolean"})
+
     add_deps("cmake", "libogg")
 
+    on_fetch(function (package, opt)
+        if opt.system then
+            local vorbis = package:find_package("vorbis", opt)
+            if not vorbis then
+                return
+            end
+            local result = table.copy(vorbis)
+            if result.includedirs then
+                result.sysincludedirs = result.includedirs
+                result.includedirs = nil
+            end
+
+            if package:config("vorbisenc") then
+                local vorbisenc = package:find_package("vorbisenc", opt)
+                if not vorbisenc then
+                    return
+                end
+
+                result.sysincludedirs = table.join(vorbisenc.sysincludedirs or vorbisenc.includedirs, result.sysincludedirs)
+                result.linkdirs = table.join(vorbisenc.linkdirs, result.linkdirs)
+                result.links = table.join(vorbisenc.links, result.links)
+            end
```

Comment:
You can refer

https://github.com/xmake-io/xmake-repo/blob/30e6efbbfb178d285d7031d25222bd7ee0898dbc/packages/b/brotli/xmake.lua#L27-L44

Reply:
I will consider add a helper function to simply merge multi-package results.